src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java

```diff
@@ -1169,5 +1209,39 @@ protected boolean isInvalidSample(double sampleRate) {
         return sampleRate != 1 && ThreadLocalRandom.current().nextDouble() > sampleRate;
     }
 
+    protected boolean isOriginDetectionEnabled(String containerID, boolean originDetectionEnabled, boolean hasEntityID) {
```

Comment:
```suggestion
    private boolean isOriginDetectionEnabled(String containerID, boolean originDetectionEnabled, boolean hasEntityID) {
```

Reply:
It's `protected` not `private` to allow testing it in `NonBlockingStatsDClientTest`. Any suggestions on how to test private methods?

Reply:
Ah, in this case let's make it package private (without a visibility modifier).

Reply:
done!